src/js/main.js

```diff
@@ -168,94 +170,145 @@ Site.Map = {
   handleMouseMove: function(event) {
     var _this =  this;
 
-    var posX = event.clientX;
-    var posY = event.clientY;
-
-    _this.noPan();
-
-    if (posY > _this.panZones.up.max && posY < _this.panZones.down.min && posX > _this.panZones.left.max && posX < _this.panZones.right.min) { // not in a zone
-      _this.panning = false;
-    } else {
-      _this.panning = true;
-
-      // Check inside which zones
-      if (posY >= _this.panZones.up.min && posY <= _this.panZones.up.max) { //up
-        _this.pan.up = true;
-      }
-
-      if (posY >= _this.panZones.down.min && posY <= _this.panZones.down.max) { //down
-        _this.pan.down = true;
-      }
+    // Save mouse position
+    _this.mouse.x = event.clientX;
+    _this.mouse.y = event.clientY;
 
-      if (posX >= _this.panZones.left.min && posX <= _this.panZones.left.max) { //left
-        _this.pan.left = true;
-      }
+    if (!_this.isInsidePanZone()) { // If outside pan zones
+      _this.stopPanning(); // Stop animation
+    } else if(!_this.panning) { // If inside and not panning already
+      _this.triggerPanning(); // Trigger animation
+    }
 
-      if (posX >= _this.panZones.right.min && posX <= _this.panZones.right.max) { //right
-        _this.pan.right = true;
-      }
+  },
 
-      _this.triggerAnimation();
-    }
+  stopPanning: function() {
+    var _this =  this;
 
+    _this.panning = false;
   },
 
-  noPan: function() {
+  // (Bool) check if mouse is inside any pan zone
+  isInsidePanZone: function() {
     var _this =  this;
 
-    _this.pan =  {
-      up: false,
-      down: false,
-      left: false,
-      right: false,
-    };
+    var posX = _this.mouse.x;
+    var posY = _this.mouse.y;
+
+    if ( (posX >= _this.panZones.left.min && posX <= _this.panZones.left.max) || // Left
+      (posX >= _this.panZones.right.min && posX <= _this.panZones.right.max) || // Right
+      (posY >= _this.panZones.up.min && posY <= _this.panZones.up.max) || // Up
+      (posY >= _this.panZones.down.min && posY <= _this.panZones.down.max) ) { //  Down
+
+        return true;
+      }
 
+    return false;
   },
 
-  triggerAnimation: function() {
+  // Trigger panning animation
+  triggerPanning: function() {
     var _this =  this;
 
-    window.requestAnimationFrame(_this.animate.bind(_this));
+    _this.panning = true;
+
+    window.requestAnimationFrame(_this.pan.bind(_this));
   },
 
-  animate: function() {
+  // Main panning logic
+  pan: function() {
     var _this =  this;
 
+    // Check is panning has been stopped; we do this here to fail faster
     if (!_this.panning) {
-      return true;
+      return false;
     }
 
-    // Get current map position
+    // Get the angle between center and mouse position
+    var angle = _this.angleFromCenter();
+
+    // Get distance between center and mouse position
+    var distance = _this.distanceFromCenter() * _this.panSpeed;
```

Comment:
the comment doesnt seem to follow the code. this gets that distance and manipulates it into a distance of travel?